core/runtime/v2/shim.go

```diff
@@ -371,8 +377,8 @@ func (s *shim) ID() string {
 	return s.bundle.ID
 }
 
-func (s *shim) Version() int {
-	return s.version
+func (s *shim) Endpoint() (string, int) {
+	return s.address, s.version
```

Comment:
nit.. seems like there is a desire to have endpoints be their own class.. but have not gone all the way with it ..